### PR TITLE
Se 403 tagged rules processing optimization by skipping empty rules

### PR DESCRIPTION
**Pull request by TJ-Everett**

### CODE REVIEW

File: src/economic/ruleProcessor/ERC20TaggedRuleProcessorFacet.sol [hunk 1]
```diff
@@ -173,11 +173,14 @@ contract ERC20TaggedRuleProcessorFacet is IRuleProcessorErrors, IInputErrors, IT
                 if (toTags[i] != "") {
                     TaggedRules.MinBalByDateRule memory minBalByDateRule = getMinBalByDateRule(ruleId, toTags[i]);
                     uint256 holdPeriod = minBalByDateRule.holdPeriod;
-                    /// first check to see if still in the hold period
-                    if ((block.timestamp - (holdPeriod * 1 hours)) < startTime) {
-                        uint256 holdAmount = minBalByDateRule.holdAmount;
+                    uint256 holdAmount = minBalByDateRule.holdAmount;
+                    /// check if holdAmount is 0, 0 means it is an empty-rule/no-rule. a holdAmount should be greater than 0
+                    if(holdAmount != 0) {
```
Comment (oscarsernarosero) [inline src/economic/ruleProcessor/ERC20TaggedRuleProcessorFacet.sol]: Is there a reason why we didn't use the `holdPeriod` that is already accessed?

**Review by oscarsernarosero — CHANGES_REQUESTED** [review]
This looks good, but we need to do the same for all tagged rules

File: src/economic/ruleProcessor/ERC20TaggedRuleProcessorFacet.sol (hunk 1 above)
Comment (TJ-Everett) [reply src/economic/ruleProcessor/ERC20TaggedRuleProcessorFacet.sol]: Great call - I will drop that back down a level and only call if we pass the holdPeriod check.

**Review by ShaneDuncan602 — APPROVED** [review]
Looks good.

**Pull request merged**